virttest/utils_test/libvirt.py

```diff
@@ -3087,7 +3087,7 @@ def connect_libvirtd(uri, read_only="", virsh_cmd="list", auth_user=None,
                                               uri, virsh_cmd, vm_name)
     # allow specific user to run virsh command
     if su_user != "":
-        command = "su %s -c '%s'" % (su_user, command)
+        command = "su %s -P -c '%s'" % (su_user, command)
```

Comment:
is capital letter 'P' or  lower case 'p' ? I assume here it should be lower case 'p'

Reply:
It is capital P to create a new pseudoterminal as per: https://man7.org/linux/man-pages/man1/su.1.html

Reply:
On my local environment, I met some issue when use su -P:
```
For more details see su(1).
[root@chunfulap2 pytestwcf]# su -P
su: invalid option -- 'P'


```

Reply:
It works for me. you should add 'su_user', right?

Reply:
> On my local environment, I met some issue when use su -P:
> 
> ```
> For more details see su(1).
> [root@chunfulap2 pytestwcf]# su -P
> su: invalid option -- 'P'
> ```

As mentioned by @Yingshun you probably missed new 'su_user' in your command. It works on my local Fedora as well as RHEL 8.4 and beaker RHEL9.0 machines.